### DISCUSSION TITLE
SP-API Listing async error codes

### DISCUSSION THREAD

**Discussion by k4nd1l** [amzn/selling-partner-api-models#4292] [Q&A] [answered]
I am trying to find out amazon error codes for async errors for listing items. Is there a documentation I could go to. 
https://developer-docs.amazon.com/sp-api/docs/listings-items-api-issues-troubleshooting 
This has the error codes I think for sync errors. Like for [MISSING_ATTRIBUTE].
Do these contain async ones too? There was an error code for example for non-white background which is not in the list, but cant remember its code.

**Comment by ritikacd** [comment, marked as answer]
Hi @k4nd1l 

The [troubleshooting guide](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-issues-troubleshooting) for Listings issues contains guidance for both synchronous and asynchronous errors. There are a few recommendations that you can use to ensure all errors are handled in your integration:
- Use [getListingItem](https://developer-docs.amazon.com/sp-api/docs/listings-items-api-v2021-08-01-reference#get-listings2021-08-01itemsselleridsku) operation to get the list of issues when the `includedData` request parameter includes `issues`.
- Subscribe to the [LISTINGS_ITEM_ISSUES_CHANGE](https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-use-case-guide#listings_item_issues_change) push notification to receive near realtime issues when recorded. `EnforcementActions` in the notification payload provides information on the impact of issues. You can optimize workflows based on enforcement actions and retrieving details from the Listings APIs based on issue impact.
- Use [VALIDATION_PREVIEW](https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#previewing-errors-for-a-listings-item) mode to preview errors and test your submissions before submitting the data in production. This mode includes synchronous and asynchronous errors, which will help you build your integration against most of the errors. 

Please share the error code or error message that you find missing in the troubleshooting guide.  

If you have any further questions or need additional assistance, feel free to ask.

Thanks,
Ritika
Solutions Architect

**Reply by k4nd1l** [reply]
thank you
